dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -160,7 +160,7 @@ object DistriOptimizer {
               val localModel = cached.localModels(i)
               localModel.training()
               val localCriterion = cached.localCriterions(i)
-              val (input, target) = tensorBuffer(i)
+              val (input, target) = MiniBatch.unapply(batchBuffer(i)).get
```

Comment:
This looks strange. unapply should use with match. If you don't use match, you needn't unapply but a custom function return a Tuple2.